theano/compile/function_module.py

```diff
@@ -483,6 +488,70 @@ def __copy__(self):
                 there.data = here.data
         return cpy
 
+    def copy(self, share_memory=False):
+        """
+        Copy this function. Copied function will have separated maker and fgraph
+        with original function. User can choose whether to separate storage by 
+        changing the share_memory arguments
+        ---------------------
+        Params:
+            share_memory -- { boolean } Default is False. When True, two function
+            share intermediate storages(storages except input and output storages)
+        ---------------------
+        Returns:
+            func -- Copied theano.Function
+        """
+
+        if not share_memory:
+            return self.__copy__()
+        else:
+            # copy SymbolocKits
+            ins, outs = copy.deepcopy([self.maker.inputs, self.maker.outputs])
+
+            # copy fgraph and get memo
+            maker = self.maker
+            fg_cpy, memo = maker.fgraph.clone_get_equiv(attach_feature=False)
+
+            # use copied ins, outs and fgraph to init a maker
+            new_maker = FunctionMaker(inputs=ins, outputs=outs, mode=maker.mode,
+                                        fgraph=fg_cpy, profile=maker.profile,
+                                        accept_inplace=maker.accept_inplace,
+                                        function_builder=maker.function_builder,
+                                        on_unused_input=maker.on_unused_input)
+
+            # construct new storage_map that map new variable to old storage
+            # so that the ensuing function shares storage with the original one
+            new_storage_map = {}
+            storage_map = self.fn.storage_map
+            for key in storage_map.keys():
+                # output_storages should not be shared
+                # if key not in self.maker.fgraph.outputs and \
+                #     memo.has_key(key):
```

Comment:
remove this useless code and comments. Otherwise, tell why you think the comment is true.